cognite/neat/rules/_analysis/_information_rules.py

```diff
@@ -0,0 +1,341 @@
+            properties defined for the child class, thus if no properties are defined for the child class,
+            it will not be included in the returned dictionary.
+        """
+        # TODO: https://cognitedata.atlassian.net/jira/software/projects/NEAT/boards/893?selectedIssue=NEAT-78
+
+        class_property_pairs = {}
+
+        for class_, properties in self.classes_with_properties(consider_inheritance).items():
+            processed_properties = {}
+            for property_ in properties:
+                if property_.property_ in processed_properties:
+                    # TODO: use appropriate Warning class from _exceptions.py
+                    # if missing make one !
+                    warnings.warn(
+                        f"Property {property_.property_} for {class_} has been defined more than once!"
+                        " Only the first definition will be considered, skipping the rest..",
+                        stacklevel=2,
+                    )
+                    continue
+
+                if (only_rdfpath and property_.rule_type == TransformationRuleType.rdfpath) or not only_rdfpath:
+                    processed_properties[property_.property_] = property_
+            class_property_pairs[class_] = processed_properties
+
+        return class_property_pairs
+
+    def class_linkage(self, consider_inheritance: bool = False) -> pd.DataFrame:
+        """Returns a dataframe with the class linkage of the data model.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Dataframe with the class linkage of the data model
+        """
+
+        class_linkage = pd.DataFrame(columns=["source_class", "target_class", "connecting_property", "max_occurrence"])
+
+        class_property_pairs = self.classes_with_properties(consider_inheritance)
+        properties = list(itertools.chain.from_iterable(class_property_pairs.values()))
+
+        for property_ in properties:
+            if property_.type_ == EntityTypes.object_property:
+                new_row = pd.Series(
+                    {
+                        "source_class": property_.class_,
+                        "connecting_property": property_.property_,
+                        "target_class": property_.value_type,
+                        "max_occurrence": property_.max_count,
+                    }
+                )
+                class_linkage = pd.concat([class_linkage, new_row.to_frame().T], ignore_index=True)
+
+        class_linkage.drop_duplicates(inplace=True)
+        class_linkage = class_linkage[["source_class", "connecting_property", "target_class", "max_occurrence"]]
+
+        return class_linkage
+
+    def connected_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
```

Comment:
Can't there be several subset of connected classes? I would expect this to return
```suggestion
    def connected_classes(self, consider_inheritance: bool = False) -> list[set[ClassEntity]]:
```